src/luau.cpp

```diff
@@ -16,3 +16,9 @@ extern "C" void zig_registerAssertionHandler() {
 extern "C" void zig_luau_free(void *ptr) {
     free(ptr);
 }
+
+#if LUA_VECTOR_SIZE == 4
+extern "C" void zig_lua_pushvector4(lua_State* L, float x, float y, float z, float w) {
```

Comment:
Work around for what seems like could be a bug in Zig build.  I was unable to get Zig to choose the 4-vector variant of lua_pushvector despite all the source being compiled with LUA_VECTOR_SIZE==4.  Maybe some sort of preprocessing is missing from @cImport.

Reply:
If this is the only solution I'm okay with it.. but that is really strange!

I'm going to take a look at this and see if I can figure out what is happening. I have a few ideas

Reply:
Yeah it's definitely an issue with cImport. I haven't found a solution yet, but I'll keep looking!

Reply:
Well I now understand the zig build system a lot more than I intended... but I figured it out!

At first I thought the `b.addConfigHeader()` function looked promising. It takes a list of config options and generates a .h file that can be included in the cImport. But it turns out this function is primarily meant for autoconf and cmake, where the generator spits out a header file that is later used in the project. Lua and Luau does have a similar concept with the luaconf.h file, but they expect you will edit the file before compiling rather than generate it. So that function doesn't work.

So then I found the `addCMacro()` function. I added the following to the build.zig and removed all of the workaround code you added and the tests worked with both 3 and 4 vectors 🥳 

```zig
ziglua.addCMacro("LUA_VECTOR_SIZE", b.fmt("{}", .{ luau_vector_size }));
```

Maybe there is another way, but I think this is currently the best way forward. The only thing that bothers me is the duplication of the flags for the library, and the zig module. Oh well.

Reply:
Fantastic find!  I tried it and it works. PR updated  for this. 

Seems like this did not affect C++ compilation, as I still had to add the -DLUA_VECTOR_SIZE in addition to the CMacro.

Reply:
> Seems like this did not affect C++ compilation, as I still had to add the -DLUA_VECTOR_SIZE in addition to the CMacro

Yeah I noticed that too. I think it is by design, because the macro applies to the module, and the lua artifact is separate from the module. Probably not much we can do there.